sdcflows/utils/wrangler.py

```diff
@@ -377,7 +406,10 @@ def find_estimators(
         ):
             try:
                 e = fm.FieldmapEstimation(
-                    fm.FieldmapFile(fmap.path, metadata=fmap.get_metadata())
+                    fm.FieldmapFile(
+                        fmap.path,
+                        metadata=_filter_metadata(fmap.get_metadata(), layout, subject),
```

Comment:
It looks like we are reading the metadata too many times right? 

https://github.com/nipreps/sdcflows/blob/08e198f1f62422d2d6274ed93a3b4ba7a78d661a/sdcflows/utils/wrangler.py#L427-L432

Then in 445 we use the `_resolve_intent()` filter, which is similar https://github.com/nipreps/sdcflows/blob/08e198f1f62422d2d6274ed93a3b4ba7a78d661a/sdcflows/utils/wrangler.py#L445

Perhaps we should try to avoid the second metadata read and use this one at `FieldmapFile` instantiation? (meaning, your fix looks good but the old code remains a bit brittle)

Reply:
We're not actually duplicating reads because of:

https://github.com/nipreps/sdcflows/blob/08e198f1f62422d2d6274ed93a3b4ba7a78d661a/sdcflows/utils/wrangler.py#L434-L437

Fieldmaps affected by this PR never reach the later code.

The issue here is that SDCflows will happily create an estimator for a "whole" fieldmap (doesn't need a bold/dwi/m0scan to compete it) that points to a B0FieldIdentifier, a subject-relative IntendedFor or a BIDS-URI. But when it's time to look up what estimator I should use for my BOLD file (`get_identifier()`), if I look for a subject-relative path, then I miss the BIDS-URIs.

So anyway, it could be that the better solution is for fMRIPrep to try `bids::$DS_RELPATH` along with `$SUB_RELPATH`, but when I started this it felt like SDCflows should expose the estimators the same way regardless.

Reply:
@oesteban Just checking back in on this comment. It says outdated because we removed the unused `layout` parameter, but the structure didn't change.